### ISSUE TITLE
`jsonapi_use_foreign_key_on_belongs_to_relationship = true` bug when belongs_to is blank 

### ISSUE THREAD

**Issue by InteNs** [rails-api/active_model_serializers#2335]
#### Expected behavior vs actual behavior
when the config option `jsonapi_use_foreign_key_on_belongs_to_relationship = true`
empty belongs_to relationships are serialized in a different way. (the ID attribute is missing)

##### Expected
```rb
{"data"=> {
  "id"=>"0",
  "type"=>"users",
  "attributes"=> {
    ...
  },
  "relationships"=> {
    "team-main"=>{"data"=>nil}
  }
}
```
##### Actual
```rb
{"data"=> {
  "id"=>"0",
  "type"=>"users",
  "attributes"=> {
    ...
  },
  "relationships"=> {
    "team-main"=>{"data"=>{"type"=>"teams"}},
  }
}
```
##### `config/initializers/active_model_serializers.rb`
```rb
ActiveModelSerializers.config.adapter = :json_api
ActiveModelSerializers.config.jsonapi_use_foreign_key_on_belongs_to_relationship = true
```

##### 'app/serializers/user_serializer.rb`
```rb
class UserSerializer < ApplicationSerializer
  belongs_to :team_main, type: :teams
end
```

#####  steps
```rb
object = User.new
ActiveModelSerializers::SerializableResource.new(object, {}).to_json
```

#### Environment

ActiveModelSerializers Version *(commit ref if not on tag)*:  `0.10.9`

Output of `ruby -e "puts RUBY_DESCRIPTION"`: `ruby 2.6.1p33 (2019-01-30 revision 66950) [x86_64-linux]`

OS Type & Version: 
Docker image: `ruby:2.6-slim`

Integrated application and version *(e.g., Rails, Grape, etc)*:  `Rails`

**Comment by bf4** [comment]
Good bug report. you interested in trying a fix?

**Comment by InteNs** [comment]
I've made a start in #2337 but i won't have time to setup an environment right now, it should be an easy fix.
later this week i could work on this but it might be handy if this is fixed sooner.

**Comment by InteNs** [comment]
fixed by #2337
